### DISCUSSION TITLE
Typescript compatibility

### DISCUSSION THREAD

**Discussion by GregKass** [mdn/mdn-community#689] [Browser compatibility data] [answered]
Is there something in the works to add Typescript to the Browser Compatibility section, next to Node.js?

Nearly 100% of my work now is in Typescript, between React and Angular applications. There are a number of items that do not currently work in the language, such as spread on a Set `[...new Set(itemsArray)]` or `Array.prototype.toSpliced()`, or these items may only be current as of some release version in the near future, and this info is essential to coding. I think we are beyond the point where Typescript is just the next fad, like coffeescript.

**Comment by pransh15** [comment]
Thanks for starting this discussion, @GregKass. I don't believe this is in the works so far.

TypeScript is maturing fast imo, and there might be a case to add it with the rate of adoption we've seen in the last few years. I'll take it to the team, they can discuss and share their thoughts.

**Comment by Josh-Cena** [comment]
What does "TypeScript compatibility" even mean? TypeScript is not a runtime and does not have support for runtime features. It provides declarations for what features might be available for a given runtime. For DOM APIs, you can check the inclusion criteria at https://github.com/microsoft/TypeScript-DOM-lib-generator. For ES APIs, it similarly must be supported in at least one engine. If there's something you find missing, you can file an issue asking for it to be included, such as https://github.com/microsoft/TypeScript/issues/57055

**Comment by pepelsbey** [comment, marked as answer]
Thank you for starting the discussion! We discussed this idea on a BCD call and came to the conclusion that it doesn’t fit the BCD project focus at the moment. If you’d like to discuss it further, I’d suggest filing an [issue in the BCD repo](https://github.com/mdn/browser-compat-data/issues). You’re also welcome to join the [MDN Discord](https://developer.mozilla.org/discord) if you’d prefer to chat with us more about it in sync. Closing it as completed for now.
